app/routes.php

```diff
@@ -455,7 +457,18 @@ function ( Application $app, Request $request ) use ( $ffFactory ) {
 	}
 );
 
-$app->get( '/', function ( Application $app ) {
+$app->get( '/', function ( Application $app, Request $request ) {
+	$app['session']->set( 'piwikTracking', array_filter(
+		[
+			'paymentType' => $request->get( 'zahlweise', '' ),
+			'paymentAmount' => $request->get( 'betrag', '' ),
+			'paymentInterval' => $request->get( 'periode', '' )
+		],
+		function ( string $value ) {
+			return $value !== '';
```

Comment:
Please replace with 

    return $value !== '' && strlen($value) < 20;

Otherwise a potential DDOS opportunity would be opened up by posting "War and Peace" into the three fields and generating large session files rapidly, filling up server disk space.

20 characters should be enough for large sums of money ;-)